### PR TITLE
Feature/mega emoji

### PR DESCRIPTION
**Pull request by antst**
Guys, here is my initial handling of clustering and emoji journal.

After all, for MVP I decided to not go with strictly spherical clusters, as I feel this is complication we might not need.
At the moment, cluster is a set of points for which distance from any point to any point in set is not larger than predefined MegaEmojiRadius.
in World controller there are parameters to control behavior:
```
const MegaEmojiTimeWindow = 10 * time.Second // time window for megaEmoji
const MegaEmojiRadius = 100.0
const MegaEmojiTriggerCount = 10
```

I just set random numbers, change if you wish later.

### CODE REVIEW

**Comment by antst** [comment]
Comment for @cnaize , you might find my way of removing entries from slices to be less orthodox, but it is the fasted way for those cases, which do not require to move around multiple times sub-slices.

**Comment by antst** [comment]
@RamsesDijkstra I did not test how mega-emoji triggering works. 
Just put clusterization and detection logic into 
`func (wc *WorldController) ProcessEmoji(emoji string, pos *cmath.Vec3) (bool, cmath.Vec3) `
it returns also senter-of-mass position for cluster, if you will need to use it for Unity

**Comment by antst** [comment]
@kameninc also look at it

File: pkg/cmath/math.go [hunk 1]
```diff
@@ -70,6 +70,17 @@ func Distance(x, y *Vec3) float64 {
 	return math.Sqrt(r)
 }
 
+func Distance2(x, y *Vec3) float64 {
```
Comment (cnaize) [inline pkg/cmath/math.go:73]: it's better to rename it somehow

File: internal/universe/interactions.go [hunk 2]
```diff
@@ -139,44 +132,50 @@ func (u *User) HandleEmoji(m *posbus.RelayToController) error {
...
-	var currentPond Pond
-
-	for _, p := range ponds {
-		if p.EmojiID == emojiMessage.EmojiID {
-			pondExists = true
-			// existing pond
-			currentPond = p
+func (u *User) CheckIfMegaMoji(emojiMessage *EmojiMessage, spaceID uuid.UUID, msg []byte) error {
+	if ok, pos := u.world.ProcessEmoji(emojiMessage.EmojiID, u.pos); ok {
+		emojiMessage.TargetType = 2
```
Comment (cnaize) [inline internal/universe/interactions.go:144]: it's better to use enum here

**Comment by kameninc** [comment]
Merging this, so we can test at standup. @RamsesDijkstra will optimize the code later

**Pull request merged**